src/Sulu/Bundle/CustomUrlBundle/Resources/public/js/components/webspace/settings/custom-url/input/main.js

```diff
@@ -0,0 +1,166 @@
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+define(['underscore', 'jquery', 'services/husky/util'], function(_, $, util) {
+
+    'use strict';
+
+    // Tested base-domains:
+    //  - *.sulu.io/test/*/test
+    //  - sulu.io/test/*/test
+    //  - *.sulu.io/test/*
+    //  - *.sulu.io/test
+    //  - sulu.io/test
+    //  - sulu.io
+    //  - sulu.io/*/*
+    //  - *.sulu.io/*/*
+
+    var defaults = {
+            templates: {
+                input: '<input type="text" class="form-element<% if (!!prefix) { %> prefix" data-prefix="true"<% } else { %>" data-suffix="true"<% } %>/>'
+            }
+        },
+
+        /**
+         * Returns an array of domain parts.
+         * If the domain part is empty an input should be displayed.
+         *
+         * @param {String} baseDomain
+         *
+         * @returns {String}
+         */
+        parseBaseDomain = function(baseDomain) {
+            if (!baseDomain) {
+                return [];
+            }
+
+            if (baseDomain.indexOf('*', baseDomain.length - 1) === -1) {
+                baseDomain += '/*';
+            }
+
+            return baseDomain
+                .replace(/^([^/]*)(\*)([^/]+)(.*)$/, '<span class="domain-part">$1</span>' + this.templates.input({prefix: true}) + '<span class="domain-part">$3</span><span class="domain-part">$4</span>')
```

Comment:
it is quite performant because it reuses the extracted values without returning it. actually the $1 are replacers which will be replaced by the first founded part of the regex which are the first braces.